src/test/java/com/uber/cadence/internal/replay/ReplayDeciderCacheTests.java

```diff
@@ -152,7 +152,7 @@ public void whenHistoryIsPartialAndCacheIsEmptyThenCacheEvictedExceptionIsThrown
   }
 
   @Test
-  public void evictNextWillInvalidateTheNextEntryInLineToBeEvicted() throws Exception {
+  public void evictAnyWillInvalidateAnEntryRandomlyFromTheCache() throws Exception {
```

Comment:
Why  randomly?  Is the cache not a LRU cache?

Reply:
Evict any is used for thread based eviction. In the normal mode of operation, it is lru. I.e. when elements get kicked out because of exceeding size or weight limit. However if you want to manually evict the next element, the loading cache doesn't support this. The way it was done before was by adding an element large enough to kick the next one in line out. However in this manner you have no control or visibility on which element is evicted and cannot guarantee self eviction doesn't happen.
Options to ensure this that i see is we need to keep track of insertion order but that seems like 
a duplication of effort of what the cache itself is doing (Or write our own cache). I'm not sure though ensuring this will matter on performance but i can file a task